Assets/Plugins/CountlySDK/Services/ConsentCountlyService.cs

```diff
@@ -1,47 +1,310 @@
+        }
+
+        /// <summary>
+        /// Remove consent of a group of features
+        /// </summary>
+        /// <param name="groupName">name of the consent group</param>
+        /// <returns></returns>
+        public void RemoveConsentOfFeatureGroup(string[] groupName)
+        {
+            if (!RequiresConsent) {
+                if (_config.EnableConsoleLogging) {
+                    Debug.Log("[Countly] ConsentCountlyService: Enable Consents");
+                }
+
+                return;
+            }
+
+            if (groupName == null) {
+                if (_config.EnableConsoleLogging) {
+                    Debug.Log("[Countly] ConsentCountlyService: Calling RemoveConsentOfFeatureGroup with null groupName!");
+                }
+                return;
+            }
+
+            foreach (string name in groupName) {
+                Features[] features = _countlyFeatureGroups[name];
+                foreach (Features feature in features) {
+                    RemoveConsentInternal(feature);
+                }
+            }
+
+            NotifyListeners();
+        }
+        #endregion
+
+        #region Helper Methods
+
+        private void GiveConsentInternal(Features feature)
+        {
+            if (_countlyFeatureConsents.ContainsKey(feature)) {
+                _countlyFeatureConsents[feature] = true;
+
+            } else {
+                _countlyFeatureConsents.Add(feature, true);
+            }
+
+            if (_config.EnableConsoleLogging) {
+                Debug.Log("[Countly] Setting consent for feature: [" + feature.ToString() + "] with value: [true]");
+            }
+        }
+
+        private void GiveConsentInternal(Features[] features)
+        {
+            if (!RequiresConsent) {
+                if (_config.EnableConsoleLogging) {
+                    Debug.Log("[Countly] ConsentCountlyService: Enable Consents");
+                }
+
+                return;
+            }
+
+            if (features == null) {
+                if (_config.EnableConsoleLogging) {
+                    Debug.Log("[Countly] ConsentCountlyService: Calling GiveConsent with null features list!");
+                }
+
+                return;
+            }
+            //Remove Duplicates entries
+            features = features.Distinct().ToArray();
+
+            foreach (Features feature in features) {
+                GiveConsentInternal(feature);
+            }
+        }
+
+        private void RemoveConsentInternal(Features feature)
+        {
+            if (_countlyFeatureConsents.ContainsKey(feature)) {
+                _countlyFeatureConsents[feature] = false;
+            } else {
+                _countlyFeatureConsents.Add(feature, false);
+            }
+
+            if (_config.EnableConsoleLogging) {
+                Debug.Log("[Countly] Setting consent for feature: [" + feature.ToString() + "] with value: [false]");
+            }
+        }
+
+        private void NotifyListeners()

```

Comment:
Notify should update with the changed values and not with the current state of consent.

Reply:
it is calling with the updated values, not with the current state 
<img width="484" alt="Screenshot 2021-01-15 at 11 43 04 AM" src="https://user-images.githubusercontent.com/8924406/104690815-eeb2c880-5726-11eb-875a-1dc1beb46d25.png">

Reply:
That still has too much complexity. It should be simplified and the variable "_modifiedConsents" should be removed as it does not provide any value.